src/Traits/Resettable.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CodeIgniter\Shield\Traits;
+
+use CodeIgniter\Shield\Authentication\Authenticators\Session;
+use CodeIgniter\Shield\Models\UserIdentityModel;
+
+/**
+ * Reusable methods to help the
+ * enforcing of password resets
+ */
+trait Resettable
+{
+    /**
+     * Returns true|false based on the value of the
+     * force reset column of the user's identity.
+     */
+    public function requiresPasswordReset(): bool
+    {
+        $identity_model = model(UserIdentityModel::class);
+        $identity       = $identity_model->getIdentityByType($this, Session::ID_TYPE_EMAIL_PASSWORD);
+
+        return $identity->force_reset;
+    }
+
+    /**
+     * Force password reset
+     */
+    public function forcePasswordReset(): void
+    {
+        // Do nothing if user already requires reset
+        if ($this->requiresPasswordReset()) {
+            return;
+        }
+
+        // Set force_reset to true
+        $identity_model = model(UserIdentityModel::class);
+        $identity_model->set('force_reset', 1);
+        $identity_model->where(['user_id' => $this->id, 'type' => Session::ID_TYPE_EMAIL_PASSWORD]);
+        $identity_model->update();
```

Comment:
The return value is not checked, but it seems better to change `UserIdentityModel::update()` method to throw an exception when error occurs.
So this PR is okay.

Reply:
Thank you.

Reply:
I sent PR #622